### ISSUE TITLE
[TW-1712] Command to toggle active task

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1736]
_Christiaan on 2015-10-09T07:02:04Z says:_ 

 Usually only one task can be active at a time, but this is not strictly enforced. If it was then task start could automatically stop the current active task. The "task 42 toggle" command will stop the current active task and then start the specified task. Always typing stop and then start is tedious.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-10-09T07:02:04Z
Modified: 2015-10-19T15:42:28Z
```

**Comment by taskwarrior** [comment]
_Wim Schuermann on 2015-10-09T15:09:25Z says:_ 

 This can be done through the current hook system:
[https://gist.github.com/wbsch/07edbf4b1393b1608613]

That hook will stop all active tasks before actually starting new tasks. It's inefficient and ugly as hell, but it works. :)

EDIT:
This does not add a "toggle" command - it stops all active tasks when "task FILTER start" is run. While adding commands through the use of hooks is possible, it is not currently officially supported and, you need to trust me on this, even uglier than the hook i linked to.

**Comment by taskwarrior** [comment]
_Christiaan on 2015-10-19T12:12:58Z says:_ 

 Thanks for the tip! Using the on-launch hook for this works like a charm.

**Comment by taskwarrior** [comment]
_Wim Schuermann on 2015-10-19T15:42:28Z says:_ 

 Closing as "Won't Fix" because the request can be dealt with by using hooks.

Christiaan, please note that I updated the script slightly on Paul's suggestion. You can use the new version as a drop-in replacement for the old one.

**Comment by sskras** [comment]
_Wim Schuermann on 2015-10-09T15:09:25Z says:_

> This can be done through the current hook system: [[https://gist.github.com/wbsch/07edbf4b1393b1608613](https://gist.github.com/wbsch/07edbf4b1393b1608613)]
> 
> That hook will stop all active tasks before actually starting new tasks. It's inefficient and ugly as hell, but it works. :)

Having a 3rd party repo isn't intuitive for the user.  And I need such functionality daily.

Earlier I just customized my time tracking script to stop the older started tasks by itself.  But after 4y of the use I started to search for something more elegant and found this script.

_Folks_, can it be imported into Taskwarrior repo for the easier centralized use or even further enhancements, like it's done with `on-modify.timewarrior` in the TimeWarrior repo:

https://github.com/GothenburgBitFactory/timewarrior/blob/develop/ext/on-modify.timewarrior

?
